lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  type Compatibility,
+  DescribeAddonVersionsCommand,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import * as awsEksAddonVersioning from '../../versioning/aws-eks-addon';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { EksAddonsFilter } from './schema';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  override readonly defaultVersioning = awsEksAddonVersioning.id;
+  override readonly caching = true;
+  private readonly clients: Record<string, EKSClient> = {};
+
+  constructor() {
+    super(AwsEKSAddonDataSource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AwsEKSAddonDataSource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases:${packageName}`,
+  })
+  async getReleases({
+    packageName: serializedFilter,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const res = EksAddonsFilter.safeParse(serializedFilter);
+    if (!res.success) {
+      logger.warn(
+        { err: res.error, serializedFilter },
+        'Error parsing eks-addons config.',
+      );
+      return null;
+    }
+
+    const filter = res.data;
+
+    const cmd = new DescribeAddonVersionsCommand({
+      kubernetesVersion: filter?.kubernetesVersion,
```

Comment:
```suggestion
      kubernetesVersion: filter.kubernetesVersion,
```
filter should not null or undefined here